src/main/java/com/laserfiche/repository/api/clients/impl/EntriesClientImpl.java

```diff
@@ -102,26 +99,24 @@ public CompletableFuture<ODataValueOfIListOfFieldValue> assignFieldValues(String
                     if (httpResponse.getStatus() == 429) {
                         throw new RuntimeException("Rate limit is reached.");
                     }
+                    if (httpResponse.getStatus() >= 299) {
+                        throw new RuntimeException(httpResponse.getStatusText());
+                    }
                     return httpResponse.getBody();
                 });
     }
 
     @Override
-    public CompletableFuture<CreateEntryResult> importDocument(PostEntryWithEdocMetadataRequest requestBody,
-            String repoId, Integer parentEntryId, String fileName, Boolean autoRename, String culture) {
-        Map<String, Object> queryParameters = new HashMap<>();
-        if (autoRename != null) {
-            queryParameters.put("autoRename", autoRename);
-        }
-        if (culture != null) {
-            queryParameters.put("culture", culture);
-        }
+    public CompletableFuture<CreateEntryResult> importDocument(String repoId, Integer parentEntryId, String fileName,
+            Boolean autoRename, String culture, File file, PostEntryWithEdocMetadataRequest requestBody) {
+        Map<String, Object> queryParameters = getNonNullParameters(new String[]{"autoRename", "culture"},
+                new Object[]{autoRename, culture});
+        Map<String, Object> pathParameters = getNonNullParameters(new String[]{"repoId", "parentEntryId", "fileName"},
+                new Object[]{repoId, parentEntryId, fileName});
         return httpClient
                 .post(baseUrl + "/v1/Repositories/{repoId}/Entries/{parentEntryId}/{fileName}")
-                .routeParam("repoId", repoId)
-                .routeParam("parentEntryId", String.valueOf(parentEntryId))
-                .routeParam("fileName", fileName)
                 .queryString(queryParameters)
+                .routeParam(pathParameters)
                 .contentType("application/json")
```

Comment:
does import api work when content type isn't `multipart/form-data`?

Reply:
No, it won't work. Our current tests haven't covered this yet (all existing tests pass). For this one specifically, let me update the code. The change didn't survive a earlier rebase (in the code generator project).

Reply:
Fixed it. Thanks!